crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -117,12 +118,11 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), MqttError> {
                 eprintln!("INFO: Connected");
                 client.subscribe(cmd.topic.as_str(), cmd.qos).unwrap();
             }
-            // TODO: should we keep trying to reconnect for all errors, or just
-            // if the broker isn't up and abort when e.g. we receive connection
-            // refused?
             Err(err) => {
+                if interrupted.load(Ordering::Relaxed) {
```

Comment:
Since the performance of these utilities is not at all a concern, but immediate response to the `Ctrl + C` command is desired, isn't it better to go with a stronger ordering guarantee using `Ordering::Acquire` and `Ordering::Release` so that the program closes as soon as `Ctrl + C` is pressed?

Reply:
Yeah. A user will not notice the difference. Besides, the main delay is introduced by the event loop. The break will occur only after a connection error is reported.

Reply:
The delay (of up to 1 second) between pressing `ctrl+c` is on there when the MQTT broker is not reachable, so I don't think the user will run into it very often, so it's ok from UX perspective.